interop-cats/shared/src/main/scala/zio/interop/cats.scala

```diff
@@ -385,3 +388,39 @@ final private class CatsContravariant[E, T] extends Contravariant[ZIO[*, E, T]]
   override def contramap[A, B](fa: ZIO[A, E, T])(f: B => A): ZIO[B, E, T] =
     ZIO.accessM[B](b => fa.provide(f(b)))
 }
+
+final private class NonEmptyChunkReducible extends Reducible[NonEmptyChunk] {
+
+  override def foldLeft[A, B](fa: NonEmptyChunk[A], b: B)(f: (B, A) => B): B = fa.foldLeft(b)(f)
+  override def foldRight[A, B](fa: NonEmptyChunk[A], lb: Eval[B])(f: (A, Eval[B]) => Eval[B]): Eval[B] = {
```

Comment:
This is going to have to traverse the entire chunk to build up the `Eval` data structure even though we may only need to evaluate part of it. Instead you can do:

```scala
  override def foldRight[A, B](fa: NonEmptyChunk[A], lb: Eval[B])(f: (A, Eval[B]) => Eval[B]): Eval[B] = {
    val length = fa.length
    def loop(n: Int): Eval[B] =
      if (n == length) lb
      else f(fa(n), Eval.defer(loop(n + 1)))

    Eval.defer(loop(0))
  }
```

Can make a similar change with `reduceLeftTo` below.

Reply:
@renancaju I think this comment is the only thing preventing a merge currently? ☝️

Reply:
To tell the truth, I didn't do it because I didn't fully understand Adam's suggestion. Instead of just copying and pasting what he said, I decided this feature should be in more experienced hands. Please, feel free to close this PR and implement a better one from scratch. Sorry I didn't say that before!